### PR TITLE
Provide access to transport metadata in Thrift request handlers.

### LINKED ISSUE #904: Make caller name available in server-side request object for Thrift
Protocol headers are available in raw requests so caller name is accessible there. 

Thrift requests use a different object type. `headers` there refers to the application headers, not the protocol headers. We should make the caller name available as a separate property in there.

### PR DESCRIPTION
**Pull request by abhinav**
Resolves #904.

CC @blampe @breerly @junchaowu @Raynos

### CODE REVIEW

File: python/GUIDE.rst [hunk 1]
```diff
@@ -96,6 +99,7 @@ will register handlers against. Open up ``keyvalue/server.py`` and write
 something like this:
 
 .. code-block:: python
+
```
Comment (blampe) [inline python/GUIDE.rst]: Thanks :)

**Comment by blampe** [comment]
:+1:

File: python/CHANGES.rst [hunk 2]
```diff
@@ -4,9 +4,11 @@ Changelog for tchannel.py
 0.9.2 (unreleased)
```
Comment (breerly) [inline python/CHANGES.rst]: This should be `0.10.0`.

**Comment by breerly** [comment]
Really happy with this :+1:

**Pull request merged**